internal/git/config_test.go

```diff
@@ -22,6 +22,22 @@ func TestNoRemote(t *testing.T) {
 	require.EqualError(t, err, `no remote configured to list refs from`)
 }
 
+func TestRelativeRemote(t *testing.T) {
+	ctx := context.Background()
+	testlib.Mktmp(t)
+	testlib.GitInit(t)
+	testlib.GitRemoteAddWithName(t, "upstream", "git@github.com:goreleaser/goreleaser.git")
+	_, err := git.Run(ctx, "pull", "upstream", "main")
+	require.NoError(t, err)
+	_, err = git.Run(ctx, "branch", "--set-upstream-to", "upstream/main")
+	require.NoError(t, err)
+	_, err = git.Run(ctx, "checkout", "--track", "-b", "relative_branch")
+	require.NoError(t, err)
```

Comment:
maybe also assert that `.git/config` is how we want it to be?

Reply:
I added some testing right below this.

I also changed the Git URL to be HTTPS so that it would work on pull via the CI.